src/input.ml

```diff
@@ -650,10 +651,11 @@ let di3_datastuff_really now the_ststuff conn seg _bsd_fast_path ourfinisacked f
      here because there is still urgent data to be received, but now in a future
      segment. :*)
   let data_trimmed_left_right =
-    Cstruct.sub data_trimmed_left 0 (Int.min cb.rcv_wnd (Cstruct.length data_trimmed_left))
+    let len = Int.min cb.rcv_wnd (Rope.length data_trimmed_left) in
+    Rope.sub data_trimmed_left ~off:0 ~len
   in
   let fin_trimmed =
-    if Cstruct.equal data_trimmed_left_right data_trimmed_left then
+    if Rope.length data_trimmed_left_right = Rope.length data_trimmed_left then
```

Comment:
Here this diff is related to #50, we must confirm this other PR. I can use `Rope.equal` also as we used `Cstruct.equal` before.

Reply:
would you mind to rebase now that #50 is merged?